crmprtd/wmb/normalize.py

```diff
@@ -1,70 +1,87 @@
-# Standard module
-import sys
-import os
-import csv
-
-from datetime import datetime
+# Installed libraries
+import pytz
+import logging
+import itertools
+from dateutil.parser import parse
 
 # Local
-from crmprtd.wmb import ObsProcessor, DataLogger
+from crmprtd import Row
 
-# Installed libraries
-from sqlalchemy import create_engine
-from sqlalchemy.orm import sessionmaker
 
+def normalize(file_stream):
+    log = logging.getLogger(__name__)
+    log.info('Starting WMB data normalization')
+
+    var_names = ['station_code',
+                 'weather_date',
+                 'precipitation',
+                 'temperature',
+                 'relative_humidity',
+                 'wind_speed',
+                 'wind_direction',
+                 'ffmc',
+                 'isi',
+                 'fwi',
+                 'rn_1_pluvio1',
+                 'snow_depth',
+                 'snow_depth_quality',
+                 'precip_pluvio1_status',
+                 'precip_pluvio1_total',
+                 'rn_1_pluvio2',
+                 'precip_pluvio2_status',
+                 'precip_pluvio2_total',
+                 'rn_1_RIT',
+                 'precip_RIT_Status',
+                 'precip_RIT_total',
+                 'precip_rgt',
+                 'solar_radiation_LICOR',
+                 'solar_radiation_CM3']
+
+    for row in itertools.islice(file_stream, 1, None):
+        d = {val: item
+             for val, item in zip(var_names,
+                                  row.strip().replace('"', '').split(','))}
 
-def save_file(reader, cache_dir, data):
-    # save the downloaded file
-    fname_out = os.path.join(cache_dir,
-                             'wmb_download' +
-                             datetime.strftime(datetime.now(),
-                                               '%Y-%m-%dT%H-%M-%S') +
-                             '.csv')
-    with open(fname_out, 'w') as f_out:
-        copier = csv.DictWriter(f_out, fieldnames=reader.fieldnames)
-        copier.writeheader()
-        copier.writerows(data)
+        for key, value in d.items():
```

Comment:
So this isn't necessarily wrong, but it's a little weird to iterate over everything, but skip the first two things. But I get it. I can't think of an obviously cleaner way to do it off the top of my head. But I had to read the code carefully *and* look at sample input to understand what you were doing here. Maybe add an explanatory comment at the top of the block about out the input contains a single line for each hour of observation with multiple variable quantities.